shinydashboard/lantern/global.R

```diff
@@ -0,0 +1,68 @@
+# Any code in this file is guaranteed to be called before either
+# ui.R or server.R
+
+#
+# Lantern metrics dashboard
+# This Shiny application will display metrics on FHIR endpoints as 
+# monitored by the Lantern application. 
+# 
+
+config_yaml <- yaml::read_yaml("configuration.yml")
+purrr::walk(config_yaml$libraries, library, character.only = T)
+purrr::walk(config_yaml$function_files, source)
+purrr::walk(config_yaml$module_files, source)
+
+# Load table of http response codes and descriptions
+root <- ifelse(Sys.getenv("HOME")=='/home/shiny',".","lantern")
+http_response_code_tbl <- read_csv(here(root,"http_codes.csv")) %>% mutate(code_chr=as.character(code))
+
+# we want the current set of http response codes from the endpoint monitoring
+# first get the entries from the metrics_labels table for http_request_responses
+http_response_ids <- metrics_labels %>%
+  filter(metric_name == "AllEndpoints_http_request_responses") %>%
+  select(id)
+
+# next, right_join against the value for each endpoint
+http_response_values <- metrics_values %>%
+  right_join(http_response_ids, by = c("labels_id" = "id"))
+
+# Compute the percentage of each response code for all responses received
+http_pct <- as_tibble(http_response_values %>% 
+                        mutate(code=as.character(value)) %>%
+                        group_by(labels_id,code,value) %>% 
+                        summarise(Percentage=n()) %>% 
+                        group_by(labels_id) %>% 
+                        mutate(Percentage=Percentage/sum(Percentage,na.rm = TRUE)*100)
+)
+# we want to graph all non-200 results by response code, but they need to be factors
+# so they can be shown as separate categories on the graph, rather than as a scalar value
+http_pctf <- http_pct %>% 
```

Comment:
would these globals make more sense in the functions that end up using them? It looks like several of these are only used in one place. are the expected to be used in multiple places?